app/main.py

```diff
@@ -1,34 +1,121 @@
+                 column: int,
+                 is_alive: bool = True
+                 ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple[int, int],
+                 end: tuple[int, int],
+                 is_drowned: bool = False
+                 ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = self.create_decks()
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def create_decks(self) -> list[Deck]:
+        decks = []
+        if self.start[0] == self.end[0]:
+            for col in range(self.start[1], self.end[1] + 1):
+                decks.append(Deck(self.start[0], col))
+        elif self.start[1] == self.end[1]:
+            for row in range(self.start[0], self.end[0] + 1):
+                decks.append(Deck(row, self.start[1]))
+        return decks
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self,
+                 row: int,
+                 column: int
+                 ) -> Deck | None:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+        return None
+
+    def fire(self,
+             row: int,
+             column: int
+             ) -> None:
+        deck = self.get_deck(row, column)
+        if deck:
+            deck.is_alive = False
+            self.is_drowned = all(not d.is_alive for d in self.decks)
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self,
+                 ships: list[tuple[tuple[int, int], tuple[int, int]]]
+                 ) -> None:
+        self.field = [["~" for _ in range(10)] for _ in range(10)]
+        self.ships = []
+        self.coord_to_ship = {}
+        for start, end in ships:
+            ship = Ship(start, end)
+            self.ships.append(ship)
+            for deck in ship.decks:
+                self.field[deck.row][deck.column] = "□"
+                self.coord_to_ship[(deck.row, deck.column)] = ship
+
+        self._validate_field()
+
+    def fire(self, ceil: tuple[int]) -> str:
```

Comment:
The `fire` method should accept a tuple of two integers, representing the row and column, rather than a single integer. This aligns with how coordinates are used elsewhere in the code.